src/openms/source/CHEMISTRY/Residue.cpp

```diff
@@ -58,30 +58,39 @@ namespace OpenMS
   }
 
   Residue::Residue(const String& name,
-                   const String& three_letter_code,
-                   const String& one_letter_code,
-                   const EmpiricalFormula& formula) :
+            const String& three_letter_code,
+            const String& one_letter_code,
+            EmpiricalFormula formula,
```

Comment:
you use the formula read-only.
So pass by `const&`

Reply:
done thanks

Reply:
thanks. done